tests/core/go_proto_library_importpath/bar.proto

```diff
@@ -0,0 +1,9 @@
+syntax = "proto3";
+
+package tests.core.go_proto_library_importpath.bar;
```

Comment:
Actually, the mapping from "tests.core.go_proto_library_importpath.bar" to "tests/core/go_proto_library_importpath/bar_go_proto" sounds very opinionated. Can we only infer the import path when `go_package` is set?

Reply:
The logic to infer the importpath is defined in https://github.com/bazelbuild/rules_go/blob/69d0fc88d0815bd11867ed3f26f36244faa3c06c/go/private/context.bzl#L384-L393 which I do not touch in this PR. It is basically `package/label` which seems logical within the Bazel world. This is also what we are currently using internally via a macro on top of the `go_proto_library` rule. To me, it sounds reasonable to keep the current behavior while at the same time maintaining the possibility of overwriting the importpath as before. So for existing users for which the explicit setting of importpath was obligatory, the behavior doesn't change as well.

Reply:
I see. Thanks for the explanation.